drivers/sensor/icm42688/icm42688_decoder.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_ICM42688_DECODER_H_
+#define ZEPHYR_DRIVERS_SENSOR_ICM42688_DECODER_H_
+
+#include <zephyr/drivers/sensor.h>
+
```

Comment:
I would recommend starting off by thinking of how you want the header for the data to be structured. You'll need to store in there what channels the buffer contains as well as their scale. So I think at the worst case scenario you need:
- 7 bits for the channels (bitmap)
- 2 bits for accelerometer scale (must be one of 16, 8, 4, & 2 so you can just store the enum `icm42688_accel_fs`)
- 3 bits for gyroscope scale (must be one of 2000, 1000, 500, 250, 125, 62.5, 31.25, 15.625 so you can just store the enum `icm42688_gyro_fs`)

So that puts you at 16 bit header after alignment. This means that if you were to just keep things fast you could always allocate 16 bytes (2 for the header + 2*7 for the data).
-